orc8r/cloud/go/blobstore/storage.go

```diff
@@ -32,6 +32,25 @@ type Blob struct {
 
 type Blobs []Blob
 
+// TKs converts blobs to their associated type and key.
+func (bs Blobs) TKs() []storage.TypeAndKey {
+	tks := make([]storage.TypeAndKey, 0, len(bs))
+	for _, blob := range bs {
+		tks = append(tks, storage.TypeAndKey{Type: blob.Type, Key: blob.Key})
+	}
+	return tks
+}
+
+// ByTK returns a computed view of a list of blobs as a map of
+// blobs keyed by blob TypeAndKey.
+func (bs Blobs) ByTK() map[storage.TypeAndKey]Blob {
+	ret := make(map[storage.TypeAndKey]Blob, len(bs))
+	for _, blob := range bs {
+		ret[storage.TypeAndKey{Type: blob.Type, Key: blob.Key}] = blob
+	}
+	return ret
+}
+
 func (bs Blobs) GetKeys() []string {
```

Comment:
While we're refactoring -- refactor this method to just `Keys()` to match the other methods

Generally speaking, we want to verb-ify function/method names. But for small, composable methods like this I make exception and prefer e.g. removing the `Get` for increased conciseness